File: app/src/main/java/n7/ad2/items/db/ItemsDao.java
package n7.ad2.items.db;

import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ItemsDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(ItemModel items);

    @Query("SELECT * FROM ItemModel ORDER BY name ASC")
    DataSource.Factory<Integer, ItemModel> getDataSourceItems();

    @Query("SELECT * FROM ItemModel WHERE codeName LIKE '%'||:chars||'%' ORDER BY name ASC")
    DataSource.Factory<Integer, ItemModel> getDataSourceItemsFilter(String chars);

    @Query("SELECT * FROM ItemModel")
    List<ItemModel> getAll();

}

File: feature/items/domain/impl/src/main/kotlin/n7/ad2/items/domian/internal/data/db/dao/ItemsDao.kt
package n7.ad2.items.domian.internal.data.db.dao

import androidx.room.Dao
import androidx.room.Query
import kotlinx.coroutines.flow.Flow
import n7.ad2.items.domian.internal.data.db.ItemDatabase

@Dao
interface ItemsDao {

    @Query("SELECT * FROM ItemsTable")
    fun getAllItems(): Flow<List<ItemDatabase>>

    @Query("UPDATE ItemsTable SET viewedByUser = 1 WHERE name =:name")
    fun updateItemViewedByUserField(name: String)
}
